### PR TITLE
use local repository path from maven settings

### PR DESCRIPTION
**Pull request by deki**
currently maven settings are completely ignored

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	<parent>
 		<groupId>org.springframework.cloud</groupId>
 		<artifactId>spring-cloud-build</artifactId>
-		<version>1.2.2.BUILD-SNAPSHOT</version>
```
Comment (marcingrzejszczak) [inline pom.xml]: Please go back to SNAPSHOT

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/AetherFactories.java [hunk 2]
```diff
@@ -54,7 +63,33 @@ public static RepositorySystemSession newSession(RepositorySystem system, boolea
 	}
 
 	private static String localRepositoryDirectory() {
-		return System.getProperty(MAVEN_LOCAL_REPOSITORY_LOCATION, System.getProperty("user.home") + "/.m2/repository");
+		String localRepoLocationFromSettings = settings().getLocalRepository();
```
Comment (marcingrzejszczak) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/AetherFactories.java]: This looks really nice! Could you please add a couple of tests to this?

File: pom.xml (hunk 1 above)
Comment (deki) [inline pom.xml]: ok, should be 1.2.3.BUILD-SNAPSHOT, right? 1.2.2 is already released.

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/AetherFactories.java (hunk 2 above)
Comment (deki) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/AetherFactories.java]: yes

File: pom.xml (hunk 1 above)
Comment (marcingrzejszczak) [inline pom.xml]: Yeah my bad. Please use `1.2.3.BS`

**Comment by deki** [comment]
Opened new PR #236 to 1.0.x branch.

**Pull request closed without merging**